SA3/qa_web/views.py

```diff
@@ -126,32 +126,41 @@ def questions(request):
 
 def answers(request, id_):
     q = get_object_or_404(Questions, pk=id_)
+    answer_id = [int(key.replace('select_', '')) for key in request.POST.keys() if key.startswith('select_')]
     if request.method == 'POST' and 'answer_form' in request.POST: #Update's database when somebody answers a question
         form = AnswersForm(request.POST)
-        if form.is_valid():
+        if form.is_valid(): 
             Answers.objects.create(content=request.POST['content'], owner=request.user, question=q)
     elif request.method == 'POST' and 'deselect' in request.POST:  #Update's database when somebody deselects best answer.
-        updateAnswer = Answers.objects.get(question=q, correct_answer=True)
-        updateAnswer.correct_answer = False;
-        updateAnswer.save();
-    elif request.method == 'POST': #Update's database when somebody selects a best answer.
-        answer_id = [int(key.replace('select_', '')) for key in request.POST.keys() if key.startswith('select_')]
-        if answer_id:
-            updateAnswer = Answers.objects.get(id = answer_id[0])
-            updateAnswer.correct_answer = True;
-            updateAnswer.save();
+        updateAnswer = Answers.objects.filter(question=q, correct_answer=True).last()
+        updateAnswer.correct_answer = False
+        updateAnswer.save()
+    elif answer_id: #Update's database when somebody selects a best answer.
+        updateAnswer = Answers.objects.get(id = answer_id[0])
+        updateAnswer.correct_answer = True
+        updateAnswer.save()
+    else:
+        q_answers = Answers.objects.filter(question=q, correct_answer=False)
+        q_best_answer = Answers.objects.filter(question=q, correct_answer=True)
+        for answer in q_best_answer:
+            if request.method == 'POST' and 'comment_form_'+str(answer.id) in request.POST:
+                Comments.objects.create(content=request.POST['content'], owner=request.user, question=q, answer=answer)
```

Comment:
@Lercerss can you take a look at this portion?

Reply:
This could still work, although I believe it would be best not to add the question to comments on answers since that will allow better filtering. @alvyn279 's proposed queries solve this problem beautifully.